src/States/MapViewStateDraw.cpp

```diff
@@ -92,19 +92,13 @@ void MapViewState::drawMiniMap()
 		Mine* mine = mTileMap->getTile(_mine.x(), _mine.y(), 0)->mine();
 		if (!mine) { break; } // avoids potential race condition where a mine is destroyed during an updated cycle.
 
-		if (!mine->active())
-		{
-			r.drawSubImage(mUiIcons, _mine.x() + mMiniMapBoundingBox.x() - 2, _mine.y() + mMiniMapBoundingBox.y() - 2, 0.0f, 0.0f, 7.0f, 7.0f);
-		}
-		else if (mine->active() && !mine->exhausted())
-		{
-			r.drawSubImage(mUiIcons, _mine.x() + mMiniMapBoundingBox.x() - 2, _mine.y() + mMiniMapBoundingBox.y() - 2, 8.0f, 0.0f, 7.0f, 7.0f);
-		}
-		else if (mine->exhausted())
-		{
-			r.drawSubImage(mUiIcons, _mine.x() + mMiniMapBoundingBox.x() - 2, _mine.y() + mMiniMapBoundingBox.y() - 2, 16.0f, 0.0f, 7.0f, 7.0f);
-		}
-
+		bool isMineActive = mine->active();
+		bool isMineUnused = !isMineActive;
+		bool isMineExhausted = mine->exhausted();
+		bool isMineProducing = !isMineExhausted;
+		bool isMineActiveAndProducing = isMineActive && isMineProducing;
+		float mineBeaconStatusOffsetX = isMineUnused ? 0.0f : (isMineActiveAndProducing ? 8.0f : (isMineExhausted ? 16.0f : 0.0f));
+		r.drawSubImage(mUiIcons, _mine.x() + mMiniMapBoundingBox.x() - 2, _mine.y() + mMiniMapBoundingBox.y() - 2, mineBeaconStatusOffsetX, 0.0f, 7.0f, 7.0f);
```

Comment:
I general, I like how some complicated expressions are pulled out into named variables. In this particular case, I felt it was maybe going a tad bit far. I didn't feel the names here really helped explain what were already pretty simple and easy to understand expressions.

As much as I love the ternary operator, I generally advise against nesting it multiple levels deep. The cascaded `if`/`else if` can start to be easier to read.

Embedded in this is a small change in behaviour, which is a positive. Previously, if none of the `if` cases held, the image was not drawn. Now it is always drawn, but in a modified way. That's probably a safer way to handle this situation, and should be friendlier to updates involving possible mine states.

----

I would suggest using fewer named variables here. You don't really need a new variable name just to hold the result of a named function, nor just to negate something. There are 3 conditions, so 3 boolean variables max should suffice.

Reply:
I was taught that readability trumps brevity. In this case, each variable denotes a state that the mine could be in, even though logically they only negate a previous state that negated stated doesn't mean "the opposite" of the other state. Since they all ultimately only affect one thing: the X offset of the status image; I opted for a ternary operator to reduce the use of `if/else` chains. Otherwise I would have written it like this:
```
float mineBeaconStatusOffsetX = 0.0f;
if(isMineUnused) { mineBeaconStatusOffsetX = 0.0f; }
else if(isMineActiveAndProducing) { mineBeaconStatusOffsetX = 8.0f; }
else if(isMineExhausted) { mineBeaconStatusOffsetX = 16.0f; }
else { mineBeaconStatusOffsetX = 0.0f; } //always include an else in an if-else chain!
```

Reply:
I think the issue I have is that the variable names don't tell us anything that code doesn't already tell us. Compare the left and right sides:
```cpp
bool isMineActive = mine->active();
bool isMineExhausted = mine->exhausted();
bool isMineActiveAndProducing = isMineActive && isMineProducing;
```

Reply:
True, but the variables also reduce cognitive load. I don't have to think about order of operations, the meaning of `&&`, is the use of `&&` a mistake, etc. I just have to be able to answer "is the mine active? is it exhausted? is it active and producing?"

Reply:
I've pushed a commit to address the issue, but I still think it's a bad idea. Now the function calls are being duplicated whereas before they were being called exactly once.